### PR TITLE
Added dressroom script commands to open/close Dress Room window

### PR DESCRIPTION
**Pull request by MuzTank**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** 

[//]: # (Master? Slave?)

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

#hacktoberfest

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1850)

**Comment by Emistry** [comment]
Just a suggestion, why not just merge 2 script command into 1?
```
dressroom_open(<true/false>);
```
```
- true = open dress room
- false = close dress room
```
Both of the script command are just calling the same `clif_dressroom_open` with different parameter value. 
Look redundant to create separate script command just for this purpose.

**Comment by MuzTank** [comment]
Will change it in an hour

**Comment by MuzTank** [comment]
Done @Emistry

**Comment by Asheraf** [comment]
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions.

---

*[doc/script_commands.txt, line 9896 at r3](https://reviewable.io:443/reviews/herculesws/hercules/1850#-KvS0q1pg_894XYZ57Fc:-KvS0q1pg_894XYZ57Fd:bv0hjxd) ([raw file](https://github.com/herculesws/hercules/blob/37f0df6aadc460cc729a497af438ceee1bd5235a/doc/script_commands.txt#L9896)):*
> ```
> parameter 'false' will close it. Default 'true'.
> 
> 	mes "Close this window to open the Dress Room window.";
> ```

Please follow scripting standards in documentation as well

mes("....") ;
close2();

---

*[src/map/script.c, line 23401 at r3](https://reviewable.io:443/reviews/herculesws/hercules/1850#-KvS1L7rOYk0fz76-yTX:-KvS1L7spzIPJI8o6jVf:b4yoy17) ([raw file](https://github.com/herculesws/hercules/blob/37f0df6aadc460cc729a497af438ceee1bd5235a/src/map/script.c#L23401)):*
> ```C
> 		clif_dressroom_open(sd, 1);
> 	}
> 	else {
> ```

Please follow coding style guideline https://github.com/HerculesWS/Hercules/wiki/Coding-Style

} else { 

---

*[src/map/script.c, line 23402 at r3](https://reviewable.io:443/reviews/herculesws/hercules/1850#-KvS1ZBysA_NGn8-jtFS:-KvS1ZBysA_NGn8-jtFT:b-n9v5co) ([raw file](https://github.com/herculesws/hercules/blob/37f0df6aadc460cc729a497af438ceee1bd5235a/src/map/script.c#L23402)):*
> ```C
> 	}
> 	else {
> 		if( script_getnum(st,2) )
> ```

Same as above 

if (....)

---

*[src/map/script.c, line 23410 at r3](https://reviewable.io:443/reviews/herculesws/hercules/1850#-KvS1f-gD6RJtvMO6KtG:-KvS1f-gD6RJtvMO6KtH:b-xi696p) ([raw file](https://github.com/herculesws/hercules/blob/37f0df6aadc460cc729a497af438ceee1bd5235a/src/map/script.c#L23410)):*
> ```C
> 	return true;
> #else
> 	return false;
> ```

warning dressing room available only for client versions >= 20150513

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/herculesws/hercules/1850)*

File: src/map/script.c [hunk 1]
```diff
@@ -23384,6 +23384,33 @@ BUILDIN(activatepset);
...
+BUILDIN(opendressroom)
+{
+#if PACKETVER >= 20150513
+	struct map_session_data *sd = script->rid2sd(st);
+	
+	if (sd == NULL)
+		return false;
+
+	if (!script_hasdata(st, 2)) {
+		clif_dressroom_open(sd, 1);
```
Comment (4144) [inline src/map/script.c]: Please use interfaces and not direct function calls
clif_dressroom_open -> clif->dressroom_open

File: src/map/script.c [hunk 2]
```diff
@@ -23384,6 +23384,33 @@ BUILDIN(activatepset);
...
+	}
+	else {
+		if( script_getnum(st,2) )
+			clif_dressroom_open(sd, 1);
```
Comment (4144) [inline src/map/script.c]: and here use interface

File: src/map/script.c [hunk 3]
```diff
@@ -23384,6 +23384,33 @@ BUILDIN(activatepset);
...
+		else
+			clif_dressroom_open(sd, 0);
```
Comment (4144) [inline src/map/script.c]: and here use interface

File: src/map/script.c [hunk 4]
```diff
@@ -23384,6 +23384,33 @@ BUILDIN(activatepset);
...
+	struct map_session_data *sd = script->rid2sd(st);
+	
+	if (sd == NULL)
+		return false;
+
+	if (!script_hasdata(st, 2)) {
+		clif_dressroom_open(sd, 1);
+	}
+	else {
+		if( script_getnum(st,2) )
```
Comment (mekolat) [inline src/map/script.c]:
```diff
- if( script_getnum(st,2) )
+ if (script_getnum(st, 2) == 1)
```

File: src/map/script.c [hunk 5]
```diff
@@ -23384,6 +23384,33 @@ BUILDIN(activatepset);
...
+#if PACKETVER >= 20150513
+	struct map_session_data *sd = script->rid2sd(st);
+	
+	if (sd == NULL)
+		return false;
+
+	if (!script_hasdata(st, 2)) {
+		clif_dressroom_open(sd, 1);
+	}
+	else {
```
Comment (mekolat) [inline src/map/script.c]:
```diff
- }
- else {
+ } else {
```

**Comment by MuzTank** [comment]
Will change tommorow

**Comment by mekolat** [comment]
Instead of a `true / false` you could use an enum so if we ever need to expand this command it will be easier

```c
enum dressroom_mode {
	DRESSROOM_CLOSE = 0,
	DRESSROOM_OPEN  = 1
};

BUILDIN(opendressroom) {
	struct map_session_data *sd = script->rid2sd(st);
	enum dressroom_mode mode = DRESSROOM_OPEN;
	
	if (sd == NULL) {
		return false;
	}

	if (script_hasdata(st, 2)) {
		mode = script_getnum(st, 2);
	}

	switch (mode) {
	case DRESSROOM_OPEN:
		clif->dressroom_open(sd, 1);
		break;
	case DRESSROOM_CLOSE:
		clif->dressroom_open(sd, 0);
		break;
	default:
		ShowWarning("script:opendressroom: unknown mode (%i)\n", mode);
		script_pushint(st, 0);
		return false
	}

	script_pushint(st, 1);
	return true;
}
```

```c
	script->constdb_comment("opendressroom modes");
	script->set_constant("DRESSROOM_OPEN", DRESSROOM_OPEN, false, false);
	script->set_constant("DRESSROOM_CLOSE", DRESSROOM_CLOSE, false, false);
```

**Comment by MuzTank** [comment]
Done @mekolat

File: src/map/script.c [hunk 6]
```diff
@@ -24039,6 +24080,7 @@ void script_parse_builtin(void) {
 		BUILDIN_DEF(activatepset,"i"), // Activate a pattern set [MouseJstr]
 		BUILDIN_DEF(deactivatepset,"i"), // Deactive a pattern set [MouseJstr]
 		BUILDIN_DEF(deletepset,"i"), // Delete a pattern set [MouseJstr]
+		BUILDIN_DEF(dressroom,""),
```
Comment (mekolat) [inline src/map/script.c]:
```diff
- BUILDIN_DEF(dressroom, ""),
+ BUILDIN_DEF(dressroom, "?"),
```

File: doc/script_commands.txt [hunk 7]
```diff
@@ -9884,3 +9884,23 @@ Changes the name of a unit.
 Supported Types - [ MOB | HOM | PET ].
 
 returns 1 on success, 0 on failure.
+
+---------------------------------------
+
+*dressroom({true/false});
```
Comment (mekolat) [inline doc/script_commands.txt]: You should update the documentation to reflect the changes, something like:
```
*dressroom({<mode>});

This command controls the dressing room for the attached player. If no <mode>
is passed, opens the dressing room by default.

Valid <mode> for dressroom() are:

	DRESSROOM_OPEN     opens the dressing room
	DRESSROOM_CLOSE    closes the dressing room

Example:

	dressroom(DRESSROOM_OPEN);
```

**Comment by MuzTank** [comment]
Done @mekolat

**Comment by mekolat** [comment]
Rebased, squashed, cleaned

**Comment by Asheraf** [comment]
Merged, Thank you for your contribution.

**Pull request merged**